build.py

```diff
@@ -224,14 +224,41 @@ def upload_packages(packages, nightly=False):
         else:
             print "\t - Not uploading {}, already exists.".format(p)
     print ""
-    
-def run_tests():
+
+def run_tests(parallel, timeout):
     get_command = "go get -d -t ./..."
     print "Retrieving Go dependencies...",
+    sys.stdout.flush()
     run(get_command)
     print "done."
-    print "Running tests..."
-    code = os.system("go test ./...")
+    print "Running tests:"
+    if parallel is not None:
+        print "\tParallel:", parallel
+    if timeout is not None:
+        print "\tTimeout:", timeout
+    sys.stdout.flush()
+    p = subprocess.Popen(["go", "fmt", "./..."], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = p.communicate()
+    if len(out) > 0 or len(err) > 0:
+        print "Code not formatted. Please use 'go fmt ./...' to fix formatting errors."
+        print out
+        print err
+        return False
+# GO vet for 1.4 is not working for me
+    #p = subprocess.Popen(["go", "tool", "vet", "--composites=false", "./"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Go vet 1.4 is reporting these errors. So its working but not passing the current code base which seemed odd to me.
```
vet: cluster/points_writer.go: cluster/points_writer.go:277:6: expected operand, found 'range' (and 7 more errors)
influxql/ast_test.go:314: influxql.Fields composite literal uses unkeyed fields
influxql/ast_test.go:318: influxql.Dimensions composite literal uses unkeyed fields
vet: meta/store_test.go: meta/store_test.go:1031:6: expected operand, found 'range' (and 10 more errors)
vet: services/subscriber/service.go: services/subscriber/service.go:247:6: expected operand, found 'range' (and 3 more errors)
vet: tsdb/functions.go: tsdb/functions.go:265:6: expected operand, found 'range' (and 2 more errors)
tsdb/meta_test.go:14: tsdb.SeriesIDs composite literal uses unkeyed fields
tsdb/meta_test.go:15: tsdb.SeriesIDs composite literal uses unkeyed fields
tsdb/meta_test.go:16: tsdb.SeriesIDs composite literal uses unkeyed fields
tsdb/meta_test.go:28: tsdb.SeriesIDs composite literal uses unkeyed fields
tsdb/meta_test.go:29: tsdb.SeriesIDs composite literal uses unkeyed fields
tsdb/meta_test.go:30: tsdb.SeriesIDs composite literal uses unkeyed fields
```